Landing categories

denzelchung opened the pull request:
- Add images for landing page: backend to add images parameter for /categories API with array of images
`images: ['https://.../image1.jpg', 'https://.../image2.jpg']`
- Remove and cleanup rating filter
- Add integration with backend to retrieve categories
- Add API error handling
- Add loading spinner

ooihuiying commented:
Hi @denzelchung, i have made a fair bit of changes to the code regarding the API calls. Could you kindly look at them? The website is now displaying data pulled from the api. Note that because of some blockage in the backend side, currently we are doing search by the activePage. Hence, regardless of what category is chosen, the returned results you will see are the same.

1) Could you also address the above comments? Thanks.

@siyingpoof  It would be good if you could pull the latest changes from this branch once it has been merged.

denzelchung commented on src/pages/ErrorPage/Error400.jsx:
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import PortalLayout from '../../layouts/PortalLayout';
+
+const Error400 = () => (
```
I think the current 404 page is more suited for internal usage. e.g. invalid URL then it displays "page not found".
Failed API calls should have a separate display message - "error retrieving information" or "something went wrong" kind of message instead of "page not found"

denzelchung commented:
> Hi @denzelchung, i have made a fair bit of changes to the code regarding the API calls. Could you kindly look at them? The website is now displaying data pulled from the api. Note that because of some blockage in the backend side, currently we are doing search by the activePage. Hence, regardless of what category is chosen, the returned results you will see are the same.
> 
> Could you also address the above comments before we merge the PR? Thanks.
> 
> Hi @siyingpoof It would be good if you could pull the latest changes from this branch once it has been merged.

I have addressed the comments above and pushed new code accordingly. The integration for venues is working as expected. Thanks!
